src/formats/ilbm.rs

```diff
@@ -0,0 +1,18 @@
+use crate::{
+    util::{read_u16, Endian},
+    ImageResult, ImageSize,
+};
+use std::io::{BufRead, Seek, SeekFrom};
+
+pub fn size<R: BufRead + Seek>(reader: &mut R) -> ImageResult<ImageSize> {
+    reader.seek(SeekFrom::Start(20))?;
```

Comment:
You certainly more about the format than I do, but is this always correct? Reading the file format I see these lines (emphasis mine):

> For ILBMs the BMHD (Bit Map HeaDer) chunk **and any other 'vital' chunks** must appear before the BODY chunk. Any chunks appearing after BODY are considered 'extra' and many programs will leave them unread and unchanged.

and

> The BMHD chunk specifies how the image is to be displayed and is **usually** the first chunk inside the FORM.

This leads me to believe that the width/height are not always at position 20 in the file, but are based on where in the file the BMHD chunk is located which is not guaranteed to be first.

If this is the case, it might require that this finds the BMHD chunk similar to how the JPEG/TIFF handler works. Unfortunately this means you'd need to implement something that would know how to read/skip chunks based on size instead of just doing this easy check. 

Glancing at the file in a hex editor, it seems like the BMHD chunk is followed by a size of the chunk (20 bytes), so it might be easy enough to start at index 12, grab the chunk tag + size, and if it's not BMHD skip `size` bytes. If you go with that I'd also probably want it to exit as malformed if it encounters a BODY chunk before BMHD.

Reply:
I’ve made the parser more robust to handle additional chunks occurring before the BMHD chunk